uint8 code

uint16 duration
uint16 angle
